### PR TITLE
Ensure MavenSettings is reinitialised after tests. Switch to use of static property name

### PR DESCRIPTION
**Pull request by manstis**
Following discussion on IRC and [this](https://github.com/droolsjbpm/drools/commit/7b09586871606f29d0a84e8b1fd19a04acb4d530) updated tests accordingly.

### CODE REVIEW

**Comment by manstis** [comment]
@mariofusco @psiroky Care to review?

File: guvnor-project/guvnor-project-backend/src/test/java/org/guvnor/common/services/project/backend/server/ProjectRepositoryResolverImplTest.java [hunk 1]
```diff
@@ -501,15 +515,17 @@ public void testGetRepositoriesResolvingArtifact_NewGAV_IsInstalledNotDeployed()
 
         } finally {
             if ( oldSettingsXmlPath != null ) {
-                System.setProperty( "kie.maven.settings.custom",
+                System.setProperty( MavenSettings.CUSTOM_SETTINGS_PROPERTY,
                                     oldSettingsXmlPath );
+            } else {
+                System.clearProperty( MavenSettings.CUSTOM_SETTINGS_PROPERTY );
```
Comment (psiroky) [inline guvnor-project/guvnor-project-backend/src/test/java/org/guvnor/common/services/project/backend/server/ProjectRepositoryResolverImplTest.java]: This (the 6 lines above) seems to be repeating several times in the tests. Would it make sense to move to a method?

**Comment by psiroky** [comment]
@manstis looksgood, I just made a minor comment about avoiding small duplication.

**Comment by manstis** [comment]
@psiroky Updated. Please review (again).

**Comment by psiroky** [comment]
Nice, thanks @manstis.
+1 to merge

**Pull request merged**